byron/semantics/executable-spec/src/Control/State/Transition/Trace/Generator/QuickCheck.hs

```diff
@@ -196,14 +197,27 @@ shrinkTrace
   => BaseEnv sts
   -> Trace sts
   -> [Trace sts]
-shrinkTrace baseEnv tr = interpretSTS @sts @traceGenEnv baseEnv
-    $ Trace.closure env st0 `traverse` traceSignalsShrinks
+shrinkTrace baseEnv tr =
+    interpretSTS @sts @traceGenEnv baseEnv
+    $ Trace.closure env st0 `traverse` shrunk
   where
     env = Trace._traceEnv tr
     st0 = Trace._traceInitState tr
-    signals = Trace.traceSignals Trace.OldestFirst tr
-    traceSignalsShrinks =
-      QuickCheck.shrinkList (shrinkSignal @sts @traceGenEnv) signals
+    signals = Trace.traceSignals Trace.NewestFirst tr
+
+    compareLen a b = compare (length a) (length b) -- ascending sort 
+    shrunk = sortBy compareLen (shrinkSignals signals)
```

Comment:
wouldn't `sortOn length (shrinkSignals signals)` also work?

Reply:
nice, thanks!

Reply:
So for the record, would any of you care to make an issue to remove the use of `sort` and instead swap two lines to produce sorted shrinks? (And also add a QC property to `shrinkSignals` to make sure the shrinks produced are always sorted on the list length).

```
shrinkSignals (sn:_last:[]) = [[sn]]
shrinkSignals (sn:sm:sigs) =
  [[sn]] -- a trace with only the most recent signal
  ++ ((sn:) <$> shrinkSignals (sm:sigs)) -- shrink the tail
  ++ [sn:sigs] -- discard the second most recent signal  
-- shrink to [] if there is one or no signals
shrinkSignals _  = []
```

And for the record 2, also note that the shrink strategy is too aggressive. According to @uroboros that gave you guys good results, but if you start getting only very large counterexamples you might consider being a bit more conservative in the shrinker.

Reply:
nice one! #1103

Reply:
:ok_hand: